src/ucs/sys/sys.h

```diff
@@ -636,6 +636,18 @@ unsigned long ucs_sys_get_proc_create_time(pid_t pid);
 ucs_status_t ucs_sys_get_memlock_rlimit(size_t *rlimit_value);
 
 
+/*
+ * Get effective max locked memory limit (unlimited for privileged user).
+ * In case we can't query the system for capabilities, we fallback to
+ * @ref ucs_sys_get_memlock_rlimit
+ *
+ * @param [out] rlimit_value If successful, set to the current limit value.
+ *
+ * @return UCS_OK if successful, or error status if failed.
+ */
+ucs_status_t ucs_sys_get_effective_memlock_rlimit(size_t *rlimit_value);
```

Comment:
i think we better replace ucs_sys_get_memlock_rlimit (or rename it to ucs_sys_get_memlock_limit)
since the bottom line that UCS users need to know is the effective limit.
ucs_sys_get_memlock_rlimit could be removed altogether (inlined to caller function)